server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java

```diff
@@ -90,10 +90,12 @@ private RabbitMQMailQueueItem loadItem(GetResponse response) throws MailQueue.Ma
     }
 
     private ThrowingConsumer<Boolean> ack(long deliveryTag, Mail mail) {
-        return sucess -> {
+        return success -> {
             try {
-                dequeueMetric.increment();
-                rabbitClient.ack(deliveryTag);
+                if (success) {
+                    dequeueMetric.increment();
+                    rabbitClient.ack(deliveryTag);
+                }
```

Comment:
and rollback otherwise

Reply:
Crap MailQueueContract::dequeueShouldAllowRetrieveFailItems was buggy... I solved this, and the nack case is now tested...